provisioning/device/src/CertificateInstaller.cs

```diff
@@ -1,68 +1,31 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.Devices.Shared;
-using System;
-using System.Collections.Generic;
 using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Devices.Shared;
 
 namespace Microsoft.Azure.Devices.Provisioning.Client
 {
     internal static class CertificateInstaller
     {
-        private static readonly HashSet<string> s_installedCertificates = new HashSet<string>();
-        private static readonly object s_lock = new object();
-
-        static CertificateInstaller()
-        {
-            try
-            {
-                using (var store = new X509Store(StoreName.CertificateAuthority, StoreLocation.CurrentUser))
-                {
-                    store.Open(OpenFlags.ReadOnly);
-                    foreach (X509Certificate2 certificate in store.Certificates)
-                    {
-                        s_installedCertificates.Add(certificate.Thumbprint);
-                    }
-                }
-            }
-            catch (Exception ex)
-            {
-                if (Logging.IsEnabled)
-                {
-                    Logging.Error(
-                     null,
-                     $"{nameof(CertificateInstaller)} failed to read store: {ex}.");
-                }
-            }
-        }
-
         public static void EnsureChainIsInstalled(X509Certificate2Collection certificates)
         {
             if (certificates == null)
             {
                 return;
             }
 
-            lock (s_lock)
```

Comment:
I think the lock should stay. That said, it won't fix the case where multiple processes try to do the operation at the same time.
I remember having this discussion with Schannel: sometimes the certificate storage files can get in an odd state on Windows if two processes try to add the same cert at the same time.
Likely the issue exists in Linux/OpenSSL where files are created on disk.

Ideally, customers should have the Intermediates pre-installed and not use this API. (Tool available here: https://gist.github.com/CIPop/baf530337cc6b29e3d6a88047661f69c and on Windows `certmgr.msc` can be used.)

Reply:
I assumed the lock was here because the HashMap was static. Now that we don't have that, I don't see the point.

This class is only used in 1 place, in a DPS client factory method.

Reply:
If more than one DPS clients make the call the factory method will run multiple times. The only concern is when more than one DPS clients exist and they all run the code at the same time. AFAIK, the cert installer function is not thread safe.